src/worker.rs

```diff
@@ -170,15 +170,14 @@ impl Worker {
 
     fn get_work(&mut self) -> Option<Work> {
         loop {
-            match self.queue.try_pop() {
-                Some(work) => return Some(work),
-                None => {
-                    if self.quit.load(Ordering::Relaxed) {
-                        return None;
-                    } else {
-                        thread::yield_now();
-                        // thread::sleep(time::Duration::from_millis(1));
-                    }
+            if !self.queue.is_empty() {
+                return Some(self.queue.pop().unwrap());
```

Comment:
does `.pop()` return a result?

if it returns an Option we don't need to unwrap and rewrap

if it returns a result maybe we should return None if it's `Err`

Reply:
it returns a Result, I changed the code to do a `match` on the `Ok` and the `Err`, similar to the original code :-)